game/plugin/navigation/door/src/door.plugin.kts

```diff
@@ -0,0 +1,16 @@
+
+import org.apollo.game.message.impl.ObjectActionMessage
+import org.apollo.plugin.navigation.door.Door
+import org.apollo.plugin.navigation.door.OpenDoorAction
+
+/**
+ * Hook into the [ObjectActionMessage] and listens for a supported door [GameObject]
+ */
+on { ObjectActionMessage::class }
+        .where { option == 1 }
+        .then {
+            val door: Door? = Door.find(it.world, position, id)
```

Comment:
You can do the "if null then return" stuff on the assignment line, which is quite nice to avoid boilerplate further down:

```kotlin
val door = Door.find(it.world, position, id) ?: return@then
```